tests/runtime/call

```diff
@@ -109,6 +109,24 @@ AFTER ./testprogs/syscall execve /$(python3 -c "print('X'*5555)")
 REQUIRES_FEATURE probe_read_kernel
 TIMEOUT 1
 
+NAME str_tuple_scratch_buf
+PROG BEGIN { print((1, (2, "XXXX"))) }
+ENV BPFTRACE_ON_STACK_LIMIT=0
+EXPECT (1, (2, XXXX))
+
+NAME str_scratch_buf
+PROG t:syscalls:sys_enter_execve { print(str(args.filename)) }
+ENV BPFTRACE_ON_STACK_LIMIT=0
+EXPECT /XXXX
+AFTER ./testprogs/syscall execve /$(python3 -c "print('XXXX')")
+REQUIRES_FEATURE probe_read_kernel
+TIMEOUT 1
+
+NAME fmt_str_scratch_buf
```

Comment:
Sorry, I've probably picked the wrong PR to bring this up on, but the name "fmt_str_scratch_buf" is confusing me. From playing around with it a little, it seems like it's not the format string that's being put into a scratch buf, but rather all the other arguments to `printf`. The format string never appears in BPF code IIUC.

In the case where we're keeping data on the stack, the name "printf_args" is used in place of "fmtstr", which makes more sense to me.

Reply:
technically its called fmt_str_args_scratch_buf, I just fudged the naming here: https://github.com/bpftrace/bpftrace/blob/57b355c95b5406742edeb4b99330e2253703615b/src/required_resources.h#L104

I'll fix the naming here.

I kinda like format string args to printf args but if we want to change it, I prefer we do in another PR